src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_Nullable.cs

```diff
@@ -0,0 +1,760 @@
+            // PROTOTYPE(NullableReferenceTypes): Use AssertNullableAttribute(method.Parameters[0].GetAttributes()). 
+            AssertAttribute(comp);
+        }
+
+        [Fact]
+        public void EmitAttribute_LocalFunctionReturnType()
+        {
+            var source =
+@"class C
+{
+    static void M()
+    {
+        object?[] L() => throw new System.NotImplementedException();
+        L();
+    }
+}";
+            CompileAndVerify(
+                source,
+                parseOptions: TestOptions.Regular8,
+                options: TestOptions.DebugDll.WithMetadataImportOptions(MetadataImportOptions.All),
+                symbolValidator: module =>
+                {
+                    var method = module.ContainingAssembly.GetTypeByMetadataName("C").GetMethod("<M>g__L|0_0");
+                    AssertNullableAttribute(method.GetReturnTypeAttributes());
+                });
+        }
+
+        [Fact]
+        public void EmitAttribute_LocalFunctionParameters()
+        {
+            var source =
+@"class C
+{
+    static void M()
+    {
+        void L(object? x, object y) { }
+        L(null, 2);
+    }
+}";
+            CompileAndVerify(
+                source,
+                parseOptions: TestOptions.Regular8,
+                options: TestOptions.DebugDll.WithMetadataImportOptions(MetadataImportOptions.All),
+                symbolValidator: module =>
+                {
+                    var method = module.ContainingAssembly.GetTypeByMetadataName("C").GetMethod("<M>g__L|0_0");
+                    AssertNullableAttribute(method.Parameters[0].GetAttributes());
+                    AssertNoNullableAttribute(method.Parameters[1].GetAttributes());
+                });
+        }
+
+        [Fact]
+        public void UseSiteError_LambdaReturnType()
+        {
+            var source0 =
+@"namespace System
+{
+    public class Object { }
+    public abstract class ValueType { }
+    public struct Void { }
+    public struct Boolean { }
+    public struct IntPtr { }
+    public class MulticastDelegate { }
+}";
+            var comp0 = CreateCompilation(source0);
+            comp0.VerifyDiagnostics();
+            var ref0 = comp0.EmitToImageReference();
+
+            var source =
+@"delegate T D<T>();
+class C
+{
+    static void F<T>(D<T> d)
+    {
+    }
+    static void G(object o)
+    {
+        F(() =>
+        {
+            if (o != new object()) return o;
+            return null;
+        });
+    }
+}";
+            var comp = CreateCompilation(
+                source,
+                references: new[] { ref0 },
+                parseOptions: TestOptions.Regular8);
+            comp.VerifyEmitDiagnostics(
+                // error CS0518: Predefined type 'System.Attribute' is not defined or imported
+                Diagnostic(ErrorCode.ERR_PredefinedTypeNotFound).WithArguments("System.Attribute").WithLocation(1, 1),
```

Comment:
This is probably not a new issue, or related to nullable feature, but I wonder if we could tie this diagnostic to the first syntax that triggered it.
Should I file a follow-up issue?

Reply:
The two errors in the test are use-site errors when failing to create two distinct synthesized attribute types: `NullableAttribute` and `EmbeddedAttribute`. In this case, the use-site errors happen to be the same error. We will only generate one error per synthesized attribute type declaration for a single compilation though.